scripts/update-lockfile.lua

```diff
@@ -5,45 +5,30 @@ local util = require('nvim-treesitter.util')
 
 -- Load previous lockfile
 local filename = require('nvim-treesitter.install').get_package_path('lockfile.json')
-local lockfile = vim.json.decode(util.read_file(filename)) --[[@as table<string,{revision:string}>]]
+local old_lockfile = vim.json.decode(util.read_file(filename)) --[[@as table<string,{revision:string}>]]
 
----@type string?
-local skip_lang_string = os.getenv('LOCKFILE_SKIP')
-local skip_langs = skip_lang_string and vim.split(skip_lang_string, ',') or {}
-vim.print('Skipping languages: ', skip_langs)
+---@type table<string,{revision:string}>
+local new_lockfile = {}
 
-local sorted_parsers = {} --- @type {name:string,parser:ParserInfo}[]
-for k, v in pairs(require('nvim-treesitter.parsers').configs) do
-  table.insert(sorted_parsers, { name = k, parser = v })
-end
-table.sort(sorted_parsers, function(a, b)
-  return a.name < b.name
-end)
+local parsers = require('nvim-treesitter.parsers').configs
 
-local jobs = {} --- @type table<{name:string,parser:ParserInfo},SystemObj>
+local jobs = {} --- @type table<string,SystemObj>
 
 -- check for new revisions
-for i, v in ipairs(sorted_parsers) do
-  -- keep spawning jobs until we hit EMFILE
-  local ok, err = pcall(function()
-    if vim.list_contains(skip_langs, v.name) or not v.parser.install_info then
-      print('Skipping ' .. v.name)
-      return
-    end
-
-    jobs[v] = vim.system({ 'git', 'ls-remote', v.parser.install_info.url })
-  end)
-
-  if not ok or i == #sorted_parsers then
-    if err and not err:match('EMFILE') then
-      error(err)
-    end
+for k, p in pairs(parsers) do
+  if not p.install_info then
+    print('Skipping ' .. k)
+    new_lockfile[k] = old_lockfile[k]
```

Comment:
These parsers don't have a lockfile entry, so they're just skipped completely.

(We'll need to do this only for parsers that are skipped because `p.tier == 4`, if we follow my plan.)